lib/gollum/assets.rb

```diff
@@ -4,20 +4,21 @@ module Precious
   module Assets
     MANIFEST = %w(app.js editor.js mermaid.js app.css criticmarkup.css fileview.css ie7.css print.css *.png *.jpg *.svg *.eot *.ttf)
     ASSET_URL = 'gollum/assets'
-    JS_COMPRESSOR = :uglify unless defined?(JS_COMPRESSOR)
 
     def self.sprockets(dir = File.dirname(File.expand_path(__FILE__)))
       env = Sprockets::Environment.new
 
-      env.append_path ::File.join(dir, '../../node_modules')
+      env.append_path defined?(::Precious::Assets::NODE_MODULES) ?
+        ::Precious::Assets::NODE_MODULES :
+        ::File.join(dir, '../../node_modules')
 
       env.append_path ::File.join(dir, 'public/gollum/javascript')
       env.append_path ::File.join(dir, 'public/gollum/stylesheets/')
 
       env.append_path ::File.join(dir, 'public/gollum/images')
       env.append_path ::File.join(dir, 'public/gollum/fonts')
 
-      env.js_compressor  = Precious::Assets::JS_COMPRESSOR unless Precious::App.development?
+      env.js_compressor  = ::Precious::Assets::JS_COMPRESSOR if defined?(::Precious::Assets::JS_COMPRESSOR)
```

Comment:
A behavioral change here: whether `uglifier` will be used only depends on whether `Precious::Assets::JS_COMPRESSOR` is defined, and not on the `RACK_ENV`. So a user can enable uglifier in any `RACK_ENV`, but by default it is turned off in any mode. We enable it ourselves in the `precompile` task in our `Rakefile`.